htlcswitch/link.go

```diff
@@ -1150,6 +1228,132 @@ func (l *channelLink) htlcManager() {
 				"PendingLocalUpdateCount")
 		}
 
+		// If we are attempting to shutdown the link to cooperatively
+		// close and the channel is clear of our pending updates and we
+		// haven't already sent a Shutdown, we'll send one now.
+		if (l.shutdownInit || l.shutdownReceived) && !l.shutdownSent &&
```

Comment:
this func is growing too long, can we make the code here into a method?

Reply:
this is not addressed. Also I think it's probably easier if we could map out the logic as,
1. check `!l.shutdownSent && PendingLocalUpdateCount() == 0`, this decides whether we want to start the shutdown message sending flow, and,
2. decide what to do based on either,
  - `l.shutdownInit`
  - `l.shutdownReceived`